packages/didcomm/src/MessageSender.ts

```diff
@@ -341,7 +337,10 @@ export class MessageSender {
       }
 
       const encryptedMessage = await this.envelopeService.packMessage(agentContext, message, keys)
-      await this.messagePickupRepository.addMessage({
+
+      // This is an internal event that will be catched by Message Pickup module and queued, in case the
+      // Agent supports it
+      this.emitQueuePackedMessageForPickupEvent(agentContext, {
```

Comment:
I generally like this new approach, but one downside is that if no message pickup module is registered the queued messages will be lost. I think if we receive a message that can't be delivered we should probably handle that in some way(maybe we can check if there's any subscribers?)

Reply:
Also, one more consideration is that we've noticed that in serverless architectures asynchronous events don't really work well. You don't know when an incoming request is done processing. 

That's why I wanted to look into reducing the number of events emitted, but I wasn't sure yet what a good approach would be. I think callbacks make sense, but is also what we had previously (we could of course have no handler by default, and the message pickup module can register it, but then we're basically back at the injection in a different way).

Thoughts?

Reply:
Say you don't like this approach without saying you don't like this approach! :laughing: 

Seriously speaking, I think we can solve this problem in different ways, and surely we can avoid using async events for that. If we leave this "Queue Transport repository" as a core feature of DIDComm, we can queue messages synchronously. Also, if we consider mediation (the concept itself, not the coordination protocols) as a core feature, we can also setup there different queue transport policies, like the one you mentioned some time ago about only queuing messages for agents we are a mediator of.

Reply:
> Say you don't like this approach without saying you don't like this approach! 😆

I really like getting rid of the injection symbols :) 

But i've also had some bad experience trying to make credo work in serverless environments and it sadly just didn't work :(